start_servers.sh

```
@@ -32,7 +32,15 @@ check_port() {
 # get_port_process outputs the PID and process name of the process listening on the specified port.
 get_port_process() {
     local port=$1
-    lsof -i :$port | grep LISTEN | awk '{print $2, $1}' | head -1
+    # Try lsof first
+    local result=$(lsof -i :$port 2>/dev/null | grep LISTEN | awk '{print $2, $1}' | head -1)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)